From 2f6d58eacce37dc908dd594e75b5e8f03f6320fd Mon Sep 17 00:00:00 2001
From: Chillbo <36623079+Ch1llb0@users.noreply.github.com>
Date: Sat, 30 Mar 2019 00:38:21 +0100
Subject: [PATCH] Skin script dialogs fix

Custom_Backup.xml:
- add new dialog prompt when skin helper service skin backup script is disabled

Custom_Customization.xml:
- fix textbox label

SkinSettings.xml:
- open new backup dialog when skin helper service skin backup add-on is disabled

Changelog.md:
- update changelog
---
 16x9/Custom_Backup.xml        | 75 +++++++++++++++++++++++++++++++++++
 16x9/Custom_Customization.xml |  2 +-
 16x9/SkinSettings.xml         |  2 +-
 Changelog.md                  |  3 ++
 4 files changed, 80 insertions(+), 2 deletions(-)
 create mode 100644 16x9/Custom_Backup.xml

diff --git a/16x9/Custom_Backup.xml b/16x9/Custom_Backup.xml
new file mode 100644
index 0000000..8fc2afd
--- /dev/null
+++ b/16x9/Custom_Backup.xml
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<window type="dialog" id="1104">
+	<!-- yesnodialog -->
+	<include>WindowFadeAnimation</include>
+	<defaultcontrol always="true">10</defaultcontrol>
+	<controls>
+
+		<!-- Background -->
+		<include>WindowFadeBackgroundImage</include>
+
+		<include content="Time">
+			<param name="heading">Enable add-on</param>
+		</include>
+
+		<!-- Heading -->
+		<control type="label" id="1">
+			<left>-15000</left>
+			<top>40</top>
+			<width>600</width>
+			<height>48</height>
+			<align>center</align>
+			<font>Font48</font>
+			<textcolor>$VAR[TextColor1]</textcolor>
+		</control>
+
+		<!-- Details -->
+		<control type="group">
+			<control type="textbox">
+				<centerleft>50%</centerleft>
+				<centertop>50%</centertop>
+				<width>1880</width>
+				<height>600</height>
+				<align>center</align>
+				<aligny>center</aligny>
+				<font>Font36</font>
+				<autoscroll>False</autoscroll>
+				<textcolor>$VAR[TextColor1]</textcolor>
+				<label>Please enable the Skin Helper Service Skin Backup add-on under[CR]Settings/Add-on browser/My add-ons[CR]to be able to backup and restore skin settings.</label>
+			</control>
+		</control>
+
+		<!-- Button grouplist background -->
+		<include content="dialogButtonBackground">
+			<param name="id">9001</param>
+		</include>
+
+		<!-- Buttons -->
+		<control type="grouplist" id="9001">
+			<left>120</left>
+			<bottom>100</bottom>
+			<width>1680</width>
+			<height>66</height>
+			<orientation>horizontal</orientation>
+			<align>center</align>
+			<itemgap>30</itemgap>
+			<control type="button" id="10">
+				<width>auto</width>
+				<font>Font36</font>
+				<align>center</align>
+				<onclick>Dialog.Close(1104)</onclick>
+				<label>OK</label>
+			</control>
+			<control type="button" id="11">
+				<width>auto</width>
+				<font>Font36</font>
+				<align>center</align>
+				<onclick>Dialog.Close(1104)</onclick>
+				<onclick>ActivateWindow(addonbrowser)</onclick>
+				<label>Open Add-on browser</label>
+			</control>
+		</control>
+
+	</controls>
+
+</window>
diff --git a/16x9/Custom_Customization.xml b/16x9/Custom_Customization.xml
index 49ca6f9..ce0a1e0 100644
--- a/16x9/Custom_Customization.xml
+++ b/16x9/Custom_Customization.xml
@@ -35,7 +35,7 @@
 				<font>Font36</font>
 				<autoscroll>False</autoscroll>
 				<textcolor>$VAR[TextColor1]</textcolor>
-				<label>Please enable the Skin Helper add-on under Settings/Add-on browser/My add-ons to be able to customise the home menu.</label>
+				<label>Please enable the Skin Helper Service add-on under[CR]Settings/Add-on browser/My add-ons[CR]to be able to customise the home menu.</label>
 			</control>
 		</control>
 
diff --git a/16x9/SkinSettings.xml b/16x9/SkinSettings.xml
index e3780cd..54b5b7e 100755
--- a/16x9/SkinSettings.xml
+++ b/16x9/SkinSettings.xml
@@ -818,7 +818,7 @@
 					<height>66</height>
 					<textwidth>1120</textwidth>
 					<onclick condition="String.IsEmpty(System.AddonVersion(script.skin.helper.skinbackup)) + !System.HasAddon(script.skin.helper.skinbackup)">InstallAddon(script.skin.helper.skinbackup)</onclick>
-					<onclick condition="!String.IsEmpty(System.AddonVersion(script.skin.helper.skinbackup)) + !System.HasAddon(script.skin.helper.skinbackup)">ActivateWindow(1102)</onclick>
+					<onclick condition="!String.IsEmpty(System.AddonVersion(script.skin.helper.skinbackup)) + !System.HasAddon(script.skin.helper.skinbackup)">ActivateWindow(1104)</onclick>
 					<onclick condition="System.HasAddon(script.skin.helper.skinbackup)">RunScript(script.skin.helper.skinbackup)</onclick>
 					<label>31052</label>
 					<focusedcolor>$VAR[TextColor1]</focusedcolor>
diff --git a/Changelog.md b/Changelog.md
index 5afe98b..cddc313 100644
--- a/Changelog.md
+++ b/Changelog.md
@@ -79,6 +79,9 @@ Addon.Browser.xml:
 - add onload to show welcome screen after installing the skin from zip
 - add submenu indicator
 
+Custom_Backup.xml:
+- add new dialog prompt when skin helper service skin backup script is disabled
+
 Custom_Cache_Progress.xml:
 - replace colors by new color variables
 
